src/config/mod.rs

```diff
@@ -530,6 +514,16 @@ fn config_changed(config: cosmic_config::Config, keys: Vec<String>, state: &mut
                 state.common.config.input_devices = value;
                 update_input(state);
             }
+            "workspaces" => {
+                let value = get_config::<WorkspaceConfig>(&config, "workspaces");
+                state
+                    .common
+                    .shell
+                    .workspaces
+                    .update_amount(value.workspace_amount);
+                // TODO workspace mode
```

Comment:
Oh I was fearing the day, we would be able to change the mode at runtime. There is a completely untested `Shell::set_mode`-function. It will be fun to see how broken that is. :rofl: